bundle-size/app.js

```diff
@@ -242,15 +243,29 @@ module.exports = app => {
       },
     });
     const check = await context.github.checks.create(params);
-    await db('checks')
-        .insert({
-          head_sha: context.payload.pull_request.head.sha,
-          pull_request_id: context.payload.number,
-          installation_id: context.payload.installation.id,
-          owner: params.owner,
-          repo: params.repo,
-          check_run_id: check.data.id,
-        });
+
+    const checkRunId = check.data.id;
+    await db.transaction(trx => {
+      return trx('checks')
+          .first('head_sha')
+          .where('head_sha', headSha)
+          .then(existingRow => {
+            if (existingRow === undefined) {
+              return trx('checks').insert({
+                head_sha: headSha,
+                pull_request_id: context.payload.number,
+                installation_id: context.payload.installation.id,
+                owner: params.owner,
+                repo: params.repo,
+                check_run_id: checkRunId,
+              });
+            } else {
+              return trx('checks')
```

Comment:
Does this refresh an existing check that was initially pending?

Reply:
Yes - GitHub Checks are tied to a commit SHA, so if the same SHA gets re-reported as a new pull request, it just means that the first check was abandoned. e.g.:
first report (abandoned): https://github.com/ampproject/amphtml/pull/20613/checks?check_run_id=56942658
second report (active): https://github.com/ampproject/amphtml/pull/20613/checks?check_run_id=57323551

Note that this is for the same commit SHA, but we saved only the check_run_id of the first report, so we also only reported the results back to it, despite it being outdated